bioptim/limits/penalty.py

```diff
@@ -404,6 +404,95 @@ def minimize_com_velocity(penalty: PenaltyOption, all_pn: PenaltyNodeList, axes:
+
+            com_ddot_cx = BiorbdInterface.mx_to_cx(
+                "com_ddot", nlp.model.CoMddot, nlp.states["q"], nlp.states["qdot"], qddot
+            )
+            return com_ddot_cx
+
+        @staticmethod
+        def minimize_angular_momentum(penalty: PenaltyOption, all_pn: PenaltyNodeList, axes: Union[tuple, list] = None):
+            """
+            Adds the objective that the angular momentum of the model in the global reference frame should be minimized.
+            If no axis is specified, the three components of the angular momentum are minimized.
+            Otherwise, the projection of the angular momentum on the specified axis is minimized.
+            By default this function is quadratic, meaning that it minimizes towards zero.
+            Parameters
+            ----------
+            penalty: PenaltyOption
+                The actual penalty to declare
+            all_pn: PenaltyNodeList
+                The penalty node elements
+            axes: Union[tuple, list]
+                The axes to project on. Default is all axes
+            """
+            PenaltyFunctionAbstract.set_axes_rows(penalty, axes)
+            penalty.quadratic = True if penalty.quadratic is None else penalty.quadratic
+            nlp = all_pn.nlp
+            angular_momentum_cx = BiorbdInterface.mx_to_cx(
+                "angular_momentum", nlp.model.angularMomentum, nlp.states["q"], nlp.states["qdot"]
+            )
+            return angular_momentum_cx
+
+        @staticmethod
+        def minimize_linear_momentum(penalty: PenaltyOption, all_pn: PenaltyNodeList, axes: Union[tuple, list] = None):
+            """
+            Adds the objective that the linear momentum of the model in the global reference frame should be minimized.
+            If no axis is specified, the three components of the linear momentum are minimized.
+            Otherwise, the projection of the linear momentum on the specified axis is minimized.
+            By default this function is quadratic, meaning that it minimizes towards zero.
+            Parameters
+            ----------
+            penalty: PenaltyOption
+                The actual penalty to declare
+            all_pn: PenaltyNodeList
+                The penalty node elements
+            axes: Union[tuple, list]
+                The axes to project on. Default is all axes
+            """
+
+            PenaltyFunctionAbstract.set_axes_rows(penalty, axes)
+            penalty.quadratic = True if penalty.quadratic is None else penalty.quadratic
+
+            nlp = all_pn.nlp
+            com_velocity = BiorbdInterface.mx_to_cx(
+                "com_velocity", nlp.model.CoMdot, nlp.states["q"], nlp.states["qdot"]
+            )
+            if isinstance(com_velocity, SX):
+                mass = Function("mass", [], [nlp.model.mass().to_mx()]).expand()
+                mass = mass()["o0"]
+            else:
+                mass = nlp.model.mass().to_mx()
```

Comment:
Is this else useful? If yes, please test it, otherwise it should be removed

Reply:
Yes, it is. If one uses Sx or MX it doesn't work the same way.